libsolidity/formal/SMTChecker.cpp

```diff
@@ -78,8 +78,10 @@ void SMTChecker::analyze(SourceUnit const& _source, shared_ptr<Scanner> const& _
 
 bool SMTChecker::visit(ContractDefinition const& _contract)
 {
-	for (auto _var : _contract.stateVariables())
-		createVariable(*_var);
+	for (auto const& contract: _contract.annotation().linearizedBaseContracts)
+		for (auto var : contract->stateVariables())
+			if (var->isVisibleInDerivedContracts())
```

Comment:
Doesn't this exclude the private variables from the current contract?

Reply:
You might want to use ContractType::stateVariables

Reply:
Yeah ok, perhaps it's not that useful.

Reply:
Oh true, was thinking only about base classes. Will update

Reply:
and add a test